packages/mint-components/src/components/sqm-rewards-table/useRewardsTable.tsx

```diff
@@ -0,0 +1,268 @@
+            rewardSource
+            prettyRedeemedCredit
+            prettyAssignedCredit
+            prettyAvailableValue
+            exchangedRewardRedemptionTransaction {
+              id
+              creditRedeemed
+              prettyRedeemedCredit
+              unit
+              dateRedeemed
+            }
+            referral {
+              id
+              referrerUser {
+                id
+                firstName
+                lastName
+              }
+              referredUser {
+                id
+                firstName
+                lastName
+              }
+            }
+            rewardRedemptionTransactions {
+              data {
+                exchangedRewards {
+                  data {
+                    prettyValue
+                    type
+                    fuelTankCode
+                    globalRewardKey
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+`;
+
+export function useRewardsTable(
+  props: RewardsTable,
+  emptyElement: VNode,
+  loadingElement: VNode
+): GenericTableViewProps {
+  const user = useUserIdentity();
+  const programIdContext = useProgramId();
+  // Default to context, overriden by props
+  const programId = props.programId ?? programIdContext;
+
+  const rewardFilter = {
+    userId_eq: user?.id,
+    accountId_eq: user?.accountId,
+    // If no program ID, shows all programs
+    ...(programId
+      ? programId === "classic"
+        ? { programId_exists: false }
+        : { programId_eq: programId }
+      : {}),
+  };
+
+  const [content, setContent] = useReducer<
+    GenericTableViewProps["elements"],
+    Partial<GenericTableViewProps["elements"]>
+  >(
+    (state, next) => ({
+      ...state,
+      ...next,
+    }),
+    {
+      columns: [],
+      rows: [],
+      loading: false,
+      page: 0,
+    }
+  );
+
+  const {
+    envelope: rewardsData,
+    states,
+    callbacks,
+  } = usePaginatedQuery<Reward>(
+    GET_REWARDS,
+    (data) => data?.viewer?.rewards,
+    {
+      limit: props.perPage,
+      offset: 0,
+    },
+    {
+      rewardFilter,
+    },
+    !user?.jwt
+  );
+
+  const tick = useRerenderListener();
+  const components = useChildElements();
+
+  const data = rewardsData?.data;
+
+  async function getComponentData(components: Element[]) {
+    // filter out loading and empty states from columns array
+    const columnComponents = components.filter(
+      (component) => component.slot !== "loading" && component.slot !== "empty"
+    );
+    // get the column titles (renderLabel is asynchronous)
+    const columnsPromise = columnComponents?.map(async (c: any) =>
+      tryMethod(c, () => c.renderLabel())
+    );
+
+    // get the column cells (renderCell is asynchronous)
+    const cellsPromise = data?.map(async (r: Reward) => {
+      const cellPromise = columnComponents?.map(async (c: any) =>
+        tryMethod(c, () => c.renderCell([r], c))
```

Comment:
```suggestion
        tryMethod(c, () => c.renderCell([r], c))
```
i don't think this should be wrapping the reward in an array